anchor/common/qbft/src/lib.rs

```
@@ -1332,17 +1445,23 @@ where
         // prepare_justification: N/A
 
         // Construct unsigned round change
-        let unsigned_msg = self.new_unsigned_message(
+        match self.new_unsigned_message(
             QbftMessageType::RoundChange,
             data_hash,
             round_change_justifications,
             vec![],
-        );
-
-        // forget that we accepted a proposal
-        self.proposal_accepted_for_current_round = false;
-
-        self.message_sender.send(unsigned_msg);
+        ) {
+            Ok(unsigned_msg) => {
+                // forget that we accepted a proposal
+                self.proposal_accepted_for_current_round = false;
+                self.message_sender.send(unsigned_msg);
+            }
+            Err(e) => {
+                warn!("Failed to construct round change message: {:?}", e);
+                // Still reset the proposal accepted flag even if message construction failed
+                self.proposal_accepted_for_current_round = false;
```

Seems a bit confusing to do it here in this function. Like, why is that so?

---

We could return a proper error and do it at the caller